pennylane/ops/functions/matrix.py

```diff
@@ -27,6 +28,15 @@
 from pennylane.typing import TensorLike
 
 
+def catalyst_qjit(qnode):
+    """The ``catalyst.while`` wrapper method"""
+    try:
+        distribution("pennylane_catalyst")

```

Comment:
@paul0403 although I'm curious why the distribution check is needed. Simply checking

```python
qnode.__class__.__name__ == "QJIT" and hasattr(qnode, user_function)
```

should be sufficient, no?

Reply:
While I was keeping in line with the precedence in `draw.py`, personally I'm not that trilled about the distribution check. It creates an extra import that's a bit unnecessary; also the check is done by name, which is a bit fishy. If I were to rewrite this check myself I would just do 

```python
def catalyst_qjit(qnode):
   return isinstance(qnode, catalyst.jit.QJIT)
```

but I guess the problem with this approach is that you would then need to `import catalyst`, which makes catalyst a dependency of pennylane, which does not make sense. 

--------
On another note, the attribute in the `QJIT` class is actually called `original_function`, not `user_function`. The `user_function` attribute is `original_function` but with the autograph preprocessing finished: https://github.com/PennyLaneAI/catalyst/blob/main/frontend/catalyst/jit.py#L487

I thus propose the following:
```python
def catalyst_qjit(qnode):
   return qnode.__class__.__name__ == "QJIT" and hasattr(qnode, original_function)
```

Reply:
I'd be happy with that change.  Think it's a little bit more straight forward to understand.

Reply:
> I'd be happy with that change. Think it's a little bit more straight forward to understand.

I made this change alongside rewriting the docstring to make more sense. I kept `user_function` and did not change that to `original_function`.

Reply:
...alongside changing the corresponding section in `draw.py` as well

Reply:
makes sense to me!